Tutorial 6. Animations

Esteban82 opened the pull request:
I add the docs. I split the file that I made for gmt-examples in two.
The idea is only explain tutorial 1 and let the participants to see tutorial 2 if they want.

Original PR at https://github.com/GenericMappingTools/gmt-examples/pull/40

Esteban82 committed:
Add docs for tutorial 6

weiji14 committed:
Merge branch 'main' into Tut6_Animations

weiji14 committed:
Fix extra blankspace in book/_toc.yml

weiji14 commented on book/tut06_animation_extended.rst:
```diff
@@ -0,0 +1,422 @@
+Making animations
+-----------------
```
```suggestion
Making animations (extended)
----------------------------
```

weiji14 committed:
Use sphinx.ext.extlinks for gmt-module links

Xref https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html

weiji14 committed:
Add sphinxcontrib-youtube

Xref https://github.com/GenericMappingTools/gmt-examples/pull/37

weiji14 committed:
Add sphinx_gmt.gmtplot

Sphinx extension used to define the `gmtplot` directive that will execute the given GMT code and insert the generated figure into the document. Need to set `highlight_language: bash` otherwise there will be an `UnboundLocalError: cannot access local variable 'suffix' where it is not associated with a value`.

weiji14 committed:
Trim trailing blankspace

weiji14 reviewed (approved):
Thanks @Esteban82, this looks amazing, really like how you go through building an animation step by step from a simple frame to the final full resolution animation.

I've pushed some changes related to the Sphinx configuration to get the GMT code/plots and Youtube links to render (based on https://github.com/GenericMappingTools/gmt-examples/pull/37). Only have one suggestion below to change the title of the extended tutorial, but will merge this directly afterwards. Feel free to make any changes after this PR is merged (e.g. typos, extra notes, etc) if needed before the workshop.

weiji14 committed:
Update book/tut06_animation_extended.rst
